com.unity.perception/Editor/Validation/CharacterTooling.cs

```diff
@@ -0,0 +1,105 @@
+using System.Collections.Generic;
+using static UnityEngine.Perception.Content.CharacterValidation;
+using System.Linq;
+
+namespace UnityEngine.Perception.Content
+{
+    public class CharacterTooling : MonoBehaviour
+    {
+        public bool CharacterRequiredBones(Animator animator, out Dictionary<HumanBone, bool> failed)
+        {
+            var result = AvatarRequiredBones(animator);
+            failed = new Dictionary<HumanBone, bool>();
+
+            for (int i = 0; i < result.Count; i++)
+            {
+                var bone = result.ElementAt(i);
+                var boneKey = bone.Key;
+                var boneValue = bone.Value;
+
+                if (boneValue != true)
+                    failed.Add(boneKey, boneValue);
+            }
+
+            if (failed.Count == 0)
+                return true;
+
+            return false;
+        }
+
+        public bool CharacterPoseData(GameObject gameObject, out List<GameObject> failedGameObjects)
+        {
+            failedGameObjects = new List<GameObject>();
+
+            var componentsParent = gameObject.GetComponents<Transform>();
```

Comment:
True, it was easier for me to read to make sure I grabbed the right things from the parent and the children because of the large number of children. I can always change it